packages/core/src/app/payment/Payment/Payment.tsx

```diff
@@ -0,0 +1,432 @@
+    errorLogger: ErrorLogger;
+    isEmbedded?: boolean;
+    isUsingMultiShipping?: boolean;
+    checkEmbeddedSupport?(methodIds: string[]): void;
+    onCartChangedError?(error: CartChangedError): void;
+    onFinalize?(orderId?: number): void;
+    onFinalizeError?(error: Error): void;
+    onReady?(): void;
+    onSubmit?(orderId?: number): void;
+    onSubmitError?(error: Error): void;
+    onUnhandledError?(error: Error): void;
+}
+
+const Payment = (
+    {
+        errorLogger,
+        onFinalize = noop,
+        onFinalizeError = noop,
+        onReady = noop,
+        isEmbedded,
+        isUsingMultiShipping,
+        onCartChangedError = noop,
+        onSubmit = noop,
+        onSubmitError = noop,
+        onUnhandledError = noop,
+        checkEmbeddedSupport = noop,
+    }: PaymentProps,
+): ReactElement => {
+    const [didExceedSpamLimit, setDidExceedSpamLimit] = useState(false);
+    const [isReady, setIsReady] = useState(false);
+    const [selectedMethod, setSelectedMethodState] = useState<PaymentMethod | undefined>();
+    const [shouldDisableSubmit, setShouldDisableSubmit] = useState<Record<string, boolean>>({});
+    const [shouldHidePaymentSubmitButton, setShouldHidePaymentSubmitButton] = useState<Record<string, boolean>>({});
+    const [submitFunctions, setSubmitFunctions] = useState<Record<string, ((values: PaymentFormValues) => void) | null>>({});
+    const [validationSchemas, setValidationSchemas] = useState<Record<string, ObjectSchema<Partial<PaymentFormValues>> | null>>({});
+
+    const componentDidMountRef = useRef(false);
+    const grandTotalChangeUnsubscribeRef = useRef<(() => void) | undefined>();
+
+    const { checkoutState, checkoutService } = useCheckout();
+    const { analyticsTracker } = useAnalytics();
+    const { language } = useLocale();
+
+    const {
+        data: { getCheckout, getCustomer, getConfig, isPaymentDataRequired, getPaymentMethods },
+        errors: { getFinalizeOrderError,  getSubmitOrderError },
+        statuses: { isInitializingPayment: getIsInitializingPayment, isSubmittingOrder: getIsSubmittingOrder },
+    } = checkoutState;
+    const {
+        clearError,
+        finalizeOrderIfNeeded,
+        loadCheckout,
+        loadPaymentMethods,
+        submitOrder,
+        subscribe: checkoutServiceSubscribe,
+    } = checkoutService;
+    const checkout = getCheckout();
+    const customer = getCustomer();
+    const config = getConfig();
+
+    if(!checkout || !customer || !config) {
+        throw new Error('Checkout data is not available');
+    }
+
+    const { isStoreCreditApplied, shouldExecuteSpamCheck } = checkout;
+    const applyStoreCredit = checkoutService.applyStoreCredit;
+    const availableStoreCredit = customer.storeCredit;
+    const cartUrl = config.links.cartLink;
+    const paymentMethods = getPaymentMethods();
+    const { defaultMethod, methods } = useMemo(()=>getDefaultPaymentMethodAndFilteredMethods(checkoutState, paymentMethods),[paymentMethods]);
```

Comment:
Calculating the default method and filtering payment methods turned out to be very expensive.
They were taking up most of the cost in `mapToPaymentProps`. I’ve moved them out and added `useMemo`.
I'm happy to break `checkoutState` into the required objects, just want to confirm this structure first.